src/ext/java/org/opentripplanner/ext/fares/impl/GtfsFaresV2Service.java

```diff
@@ -237,25 +238,47 @@ private boolean matchesNetworkId(ScheduledTransitLeg leg, FareLegRule rule) {
       routesNetworkIds.contains(rule.networkId())
     );
   }
-}
 
-/**
- * @param itineraryProducts The fare products that cover the entire itinerary, like a daily pass.
- * @param legProducts       The fare products that cover only individual legs.
- */
-record ProductResult(Set<FareProduct> itineraryProducts, Set<LegProducts> legProducts) {
-  public Set<FareProduct> getProducts(Leg leg) {
-    return legProducts
-      .stream()
-      .filter(lp -> lp.leg().equals(leg))
-      .findFirst()
-      .map(l ->
-        l
-          .products()
-          .stream()
-          .map(LegProducts.ProductWithTransfer::product)
-          .collect(Collectors.toSet())
-      )
-      .orElse(Set.of());
+  private boolean matchesDistance(ScheduledTransitLeg leg, FareLegRule rule) {
+    // If no valid distance type is given, do not consider distances in fare computation
+    if (rule.distanceType() == null) {
+      return true;
+    }
+    Double minDistance = rule.minDistance() == null ? 0 : rule.minDistance();
+    Double maxDistance = rule.maxDistance() == null ? Double.POSITIVE_INFINITY : rule.maxDistance();
+
+    if (rule.distanceType() == 0) {
```

Comment:
If you use an enum you can use a switch statement here which also does the exhaustiveness check and tell you when you've introduced a new type.